src/reduction.jl

```diff
@@ -0,0 +1,174 @@
+using LinearAlgebra
+
+""" 
+    canelide(cons_laws::AbstractMatrix, species::AbstractMatrix)
+
+Check if the given species can be simultaneously elided using the given matrix of conservation laws.
+"""
+function canelide(C::AbstractMatrix, species::AbstractVector) 
+    # This amounts to checking that the transformation [ (kept) (elided) ] -> [ (kept) (conserved quantities) ]
+    # is injective, equivalently that the linear map (elided) -> (conserved) is injective,
+    # equivalently that it has full row rank
+    rank(@view C[:,species]) == length(species)  
```

Comment:
I am not sure about this. Are all the coefficients integers?

Reply:
So far yes. In any case at this stage of the code it is assumed the species are indices into the `states` vector